tests/test_req2flatpak.py

```diff
@@ -64,5 +70,133 @@ def test(self):
+            for platform in platforms
+        }
+        build_module = FlatpakGenerator.build_module(requirements, downloads)
+        self.validate_build_module(build_module)
+
+
+class SinglePlatformTest(unittest.TestCase, Req2FlatpakBaseTest):
+    """Test involving a single requirement and a single target platform."""
+
+    requirements = ["requests==2.28.1"]
+    target_platforms = ["cp310-x86_64"]
+
+    def validate_build_module(self, build_module):
+        """Validates that the build module installs requests."""
+        self.assertIn("requests", build_module["build-commands"][0])
+        self.assertIn("requests", build_module["sources"][0]["url"])
+
+
+class MultiPlatformTest(unittest.TestCase, Req2FlatpakBaseTest):
+    """Test involving a single requirement with platform-specific wheels."""
+
+    requirements = ["scikit-learn==1.1.3"]
+    target_platforms = ["cp310-x86_64", "cp310-aarch64"]
+
+    def validate_build_module(self, build_module):
+        """Validates that the build module installs scikit-learn."""
+        self.assertIn("scikit-learn", build_module["build-commands"][0])
+        self.assertTrue(build_module["sources"])
+
+        # each url must contain scikit-learn:
+        for source in build_module["sources"]:
+            self.assertIn("scikit_learn", source["url"])
+
+        # each architecture must be found in the urls:
+        for arch in ["x86_64", "aarch64"]:
+            self.assertTrue(
+                any(arch in source["url"] for source in build_module["sources"]),
+                "The architecture {arch} was not found in the build module's sources.",
+            )
+
+
+class MultiReqTest(unittest.TestCase, Req2FlatpakBaseTest):
+    """Test involving multiple requirements for multiple target platforms."""
+
+    requirements = ["numpy==1.23.4", "pandas==1.5.1"]
+    target_platforms = ["cp310-x86_64", "cp310-aarch64"]
+
+    def validate_build_module(self, build_module):
+        """Validates that the build module installs the required packages."""
+        packages = map(lambda r: r.split("==")[0].strip(), self.requirements)
+        architectures = map(lambda a: a.split("-")[1], self.target_platforms)
+
+        # each package must be found in the build command:
+        for package in packages:
+            self.assertIn(package, build_module["build-commands"][0])
+
+        # each package and architecture must be found in the urls:
+        for package, arch in product(packages, architectures):
+            self.assertTrue(
+                any(
+                    package in source["url"] and arch in source["url"]
+                    for source in build_module["sources"]
+                ),
+                f"The package {package} and architecture {arch} was not found in the build module's sources.",
+            )
+
+
+class EggTest(unittest.TestCase, Req2FlatpakBaseTest):
+    """Test involving a package that publishes eggs."""
+
+    requirements = ["beancount==2.3.5"]
```

Comment:
In the future `beancount` might switch to wheel distributions and stop publishing eggs.

Reply:
yes, eggs are legacy, and beancount will likely stop publishing eggs in the future. the already-published eggs will likely remain available. that's why I think the test case makes sense.

Reply:
> already-published eggs will likely remain available

Yeah, that makes sense.